scipy/fft/_basic_backend.py

```diff
@@ -0,0 +1,184 @@
+from scipy._lib._array_api import array_namespace, is_numpy
+from . import _pocketfft
+import numpy as np
+
+
+def arg_err_msg(param):
+    return f'Providing {param!r} is only supported for numpy arrays.'
+
+
+def _validate_fft_args(overwrite_x, workers, plan, norm):
+    if overwrite_x:
+        raise ValueError(arg_err_msg("overwrite_x"))
+    if workers is not None:
+        raise ValueError(arg_err_msg("workers"))
+    if plan is not None:
+        raise ValueError(arg_err_msg("plan"))
+    if norm is None:
+        norm = 'backward'
+    return norm
+
+
+def _execute(func_str, pocketfft_func, x, s, axes, norm, 
+             overwrite_x, workers, plan):
+    xp = array_namespace(x)
+    # pocketfft is used whenever SCIPY_ARRAY_API is not set,
+    # or x is a NumPy array or array-like.
+    # When SCIPY_ARRAY_API is set, we try to use xp.fft for CuPy arrays,
+    # PyTorch arrays and other array API standard supporting objects.
+    # If xp.fft does not exist, we attempt to convert to np and back to use pocketfft.
+    if is_numpy(xp):
+        return pocketfft_func(x, s, axes, norm=norm, overwrite_x=overwrite_x,
+                              workers=workers, plan=plan)
+
+    # Parameters which are not in the standard are not accepted past this point 
+    norm = _validate_fft_args(overwrite_x, workers, plan, norm)
+    if hasattr(xp, 'fft'):
+        xp_func = getattr(xp.fft, func_str)
+        return xp_func(x, s, axes, norm=norm)
+
+    x = np.asarray(x)
+    y = pocketfft_func(x, s, axes, norm=norm)
```

Comment:
What's the reasoning for banning the non-standard args when we coerce to a numpy array. Is this a BC break?

Reply:
I'm not sure if there's a good reason. I remember asking this question myself at some point, @rgommers can you recall a conversation about it? It seems reasonable to me to move `_validate_fft_args` inside the `hasattr(xp, 'fft')` block.`

If it is a BC break, then I don't think it's a serious one, since we don't coerce to `np` on main currently. For example, passing a `torch` tensor in gives us `AttributeError: 'torch.dtype' object has no attribute 'kind'` from `_pocketfft`. Also, without using different `uarray` backends, I don't think that we've promised users any sort of stability with using non-NumPy arrays with the SciPy backend.

Despite this, maybe there is some library which this would BC break for (even if it isn't actually in use), so if there isn't any strong reason to ban the params, then we should allow the arguments through here.

Reply:
I don't think we discussed it. This is minor and could go either way. This code path only gets hit for non-numpy arrays that do implement the array API standard but _not_ the `fft` extension. In case we keep the nonstandard kwargs here, that would break if and when the array library implements the `fft` extension in a new release. 

So I'd be inclined to leave it as is. It's not a BC break now since it's behind `SCIPY_ARRAY_API` and I suspect it's better to not risk the future break I described above.